File: December - 06/SakthiDharan01_magic_square_odd_order.py
import sys

def generate_magic_square(n: int):
    magic = [[0]*n for _ in range(n)]
    i, j = 0, n // 2
    for num in range(1, n*n + 1):
        magic[i][j] = num
        ni, nj = (i - 1) % n, (j + 1) % n
        if magic[ni][nj] != 0:
            i = (i + 1) % n
        else:
            i, j = ni, nj
    return magic

def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    if n % 2 == 0:
        print("Magic square is only possible for odd values of n.")
        return
    M = n * (n * n + 1) // 2
    square = generate_magic_square(n)
    print(f"Magic constant: {M}")
    for row in square:
        print(" ".join(str(x) for x in row))

if __name__ == "__main__":
    solve()

File: December - 06/SakthiDharan01_magic_square_odd_order.java
import java.util.*;

public class SakthiDharan01_magic_square_odd_order {
    private static int[][] generate(int n) {
        int[][] m = new int[n][n];
        int i = 0, j = n / 2;
        for (int num = 1; num <= n * n; num++) {
            m[i][j] = num;
            int ni = (i - 1 + n) % n;
            int nj = (j + 1) % n;
            if (m[ni][nj] != 0) {
                i = (i + 1) % n;
            } else {
                i = ni;
                j = nj;
            }
        }
        return m;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        if (n % 2 == 0) {
            System.out.println("Magic square is only possible for odd values of n.");
            return;
        }
        int M = n * (n * n + 1) / 2;
        int[][] square = generate(n);
        System.out.println("Magic constant: " + M);
        for (int[] row : square) {
            for (int i = 0; i < row.length; i++) {
                if (i > 0) System.out.print(" ");
                System.out.print(row[i]);
            }
            System.out.println();
        }
        sc.close();
    }
}
